packages/report-executions/index.js

```diff
@@ -0,0 +1,79 @@
+const get = require('lodash.get');
+const has = require('lodash.has');
+
+const { Execution } = require('@cumulus/api/models');
+
+/**
+ * Determine if event is an SNS event
+ *
+ * @param {Object} event - A Cloudwatch event object
+ * @returns {boolean} - True if event is an SNS event
+ */
+const isSnsEvent = (event) => event.EventSource === 'aws.sns';
+
+/**
+ * Get message from SNS record.
+ *
+ * @param {Object} record - Record from SNS event
+ * @returns {Object} - Message object from SNS record
+ */
+const getSnsMessage = (record) => JSON.parse(get(record, 'Sns.Message', '{}'));
+
+/**
+ * Determine if SNS message has an execution status.
+ *
+ * Note: Assuming CUMULUS-1394 will ensure `meta.status` is always set. Currently
+ * it is not set for running executions.
+ *
+ * @param {Object} message - Message from SNS record
+ * @returns {boolean} - true if message has an execution status
+ */
+const hasExecutionStatus = (message) => has(message, 'meta.status');
+
+/**
+ * Create/update execution record from SNS message.
+ *
+ * @param {Object} message - An execution message
+ * @returns {Promise} - Promise from execution create/update record operation
+ */
+async function handleExecutionMessage(message) {
+  const executionModel = new Execution();
+
+  if (['failed', 'completed'].includes(message.meta.status)) {
```

Comment:
Eventually, these statuses should be replaced by the actual Step Function terminal event statuses (`SUCCEEDED`, `ABORTED`, `FAILED`, `TIMED_OUT`)`, since a Step Function Cloudwatch event will be triggering the new Lambda that publishes execution information to the SNS topic (CUMULUS-1394). However, it didn't make sense to do that yet because the execution model only accepts an ENUM for `status` where the accepted values are `completed`, `failed`, and `running`.

I can note that this change should be made and the model updated as part of 1394.